### PR TITLE
Adds DynamoSingleVersionStore,  DynamoMultipleVersionStore & MemoryTypedStore companion object

### PR DESCRIPTION
**Pull request by kenoir**
For https://github.com/wellcometrust/platform/issues/3829

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+RELEASE_TYPE: patch
```
Comment (alexwlchan) [inline RELEASE.md]: I think this is a minor release.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+RELEASE_TYPE: minor
+
+Adds a:
+
+- `DynamoSingleVersionStore` for storing a single version of a thing (where new versions overwrite old versions).
```
Comment (alexwlchan) [inline RELEASE.md]: nit: I would say “replace” instead of overwrite

And possibly an example would be useful? Happy to talk through that.

Comment (kenoir) [reply RELEASE.md]: Can't do this now though - I agree docs are important here.

Comment (kenoir) [reply RELEASE.md]: To be clear - will wait to merge until we chat.

**Pull request merged**